default_validator/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+//! Main and default Iroha instruction validator.
+
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+#[cfg(not(test))]
+extern crate panic_halt;
+
+pub mod isi;
+
+use iroha_validator::{pass_conditions, prelude::*};
+
+/// Apply `callback` macro for all token types from this crate.
+///
+/// Callback technique is used because of macro expansion order. With that technique we can
+/// apply callback to token types declared in other modules.
+///
+/// # WARNING !!!
+///
+/// If you add new module with tokens don't forget to add it here!
+macro_rules! map_all_crate_tokens {
+    ($callback:ident) => {
+        $crate::isi::account::map_tokens!($callback);
+        $crate::isi::asset::map_tokens!($callback);
+        $crate::isi::asset_definition::map_tokens!($callback);
+        $crate::isi::domain::map_tokens!($callback);
+        $crate::isi::parameter::map_tokens!($callback);
+        $crate::isi::peer::map_tokens!($callback);
+        $crate::isi::role::map_tokens!($callback);
+        $crate::isi::trigger::map_tokens!($callback);
+        $crate::isi::validator::map_tokens!($callback);
+    };
+}
+
+pub(crate) use map_all_crate_tokens;
+
+/// Validate operation.
+#[entrypoint(params = "[authority, operation]")]
+pub fn validate(
+    authority: <Account as Identifiable>::Id,
+    operation: NeedsPermissionBox,
+) -> Verdict {
+    match operation {
+        NeedsPermissionBox::Transaction(transaction) => validate_and_execute_transaction(
+            &authority,
+            &transaction,
+            validate_and_execute_instruction,
+            validate_query,
+        ),
+        NeedsPermissionBox::Instruction(instruction) => {
+            validate_and_execute_instruction(&authority, &instruction, validate_query)
+        }
+        NeedsPermissionBox::Query(query) => validate_query(&authority, query),
+    }
```

Comment:
```suggestion
    match operation {
        NeedsPermissionBox::Instruction(instruction) => {
            validate_and_execute_instruction(&authority, &instruction, validate_query)
        }
        NeedsPermissionBox::Query(query) => validate_query(&authority, query),
    }
```

If you split the paths for transactions vs instruction/query execution as suggested here [this](https://github.com/hyperledger/iroha/pull/3329#discussion_r1153209878) then `NeedsPermissionBox` should only contain instruction/query. A smart contract can never contain transaction, it is transaction itself

Reply:
In DM we decided to move it to #3371 , I close this